### DISCUSSION TITLE
Remove a node from teleport cluster

### DISCUSSION THREAD

**Discussion by Reab0** [gravitational/teleport#7564] [Q&A] [answered]
Hi, 

How can I remove nodes from teleport cluster?

**Comment by webvictim** [comment, marked as answer]
If you just shut down the Teleport process on the node, it will stop checking in with the Teleport cluster and disappear from `tsh ls` and the web UI after around 10-15 minutes.

If you need to remove all records of the node, you can run `tctl get nodes` on your auth server, look for the UUID corresponding to your node's hostname and then `tctl rm <uuid>`.

**Reply by ap-wtioit** [reply]
with teleport 16.5.3 i needed (for openssh agentless nodes):
```bash
tctl nodes ls # uuid is easier to spot here
tctl rm nodes/<uuid>
````
